### ISSUE TITLE
The output seems to can't represent some output

### ISSUE THREAD

**Issue by XCI9** [liangrj2014/ISPD24_contest#8]
For the given example in spec:
```
Net0
{
0 0 1 1 metal1
0 0 1 3 metal2
0 2 4 3 metal3
3 2 4 4 metal2
3 3 4 4 metal1
}
```
![image](https://github.com/liangrj2014/ISPD24_contest/assets/71249961/70e928c8-7fe4-4c3d-a8de-52f1fa2ad865)

According to the solution, it seems that the solution is output all covered Gcell but not all used edges since (0, 2) should have a via but  the solution does not cover it.

For the following net, red means wires(same as example in spec, horizontal in metal3 and vertical in metal2), blue means pins, green means vias, note that in (1, 2), I don't want to have a via there since it is congestion there. But since Gcell (1, 2, metal2) and (1, 2, metal3) are adjecent Gcell, it will be viewed as a via there, which is not what I want.
![image](https://github.com/liangrj2014/ISPD24_contest/assets/71249961/15b429e5-d114-4d90-bb01-bd2f37735d71)
Is there any way to solve this?

If I have misunderstood anything, please let me know.
Thanks.

**Comment by liangrj2014** [comment]
You are right! We plan to update the output format and the evaluator to accommodate it. To be specific, the output formate will also indicate the location of vias. Here is an example:

Net0

{

### x_l y_l z_l x_h y_h z_h ###
 
 0 0 0 0 0 0

 0 0 0 0 0 1 # a via

 0 0 1 0 2 1

 0 2 1 0 2 2 # a via

...

}
where each row (x_l y_l z_l x_h y_h z_h) describes a line/rectangle in the 3D GCell graph, spanning from (x_l, y_l, z_l) to (x_h, y_h, z_h).

We will release the updated output format soon. And our evaluator will also be updated accordingly. Thanks!

**Comment by cai-zhi-jie** [comment]
a few minor questions:
1. is the x_h, y_h in the updated output format is included int the solution, as I see, in the current evaluator and contest introduction, x_h and y_h is not included as shows follow:
![image](https://github.com/liangrj2014/ISPD24_contest/assets/73417581/4955dc91-a5e5-4855-9384-142a195e7757)
![image](https://github.com/liangrj2014/ISPD24_contest/assets/73417581/e778824d-99ab-4cb4-bd4a-19472d9e8260)
but since the example you gave include "0 0 0 0 0 0", so I assume the x_h and y_h will be included in the updated output format?
2. can the via go through multiple layers in the update output format? for example "0 0 0 0 0 4"
3. will the update announced in some place for us to catch up in time? (including the release/update of benchmark, the update of evaluate script/metric, the update of input/output format and so on)

**Comment by liangrj2014** [comment]
Thanks a lot for your questions!

1. You are right.  The x_h and y_h will be included in the updated output format.
2. Yes, it can.
3. We are actively working on these. We expect to release the updated (hopefully final version) of evaluate script/metric and input/output format by the end of this week. And more benchmarks will be released in the next week, including those with up to 50M cells

Really appreciate your questions and patience!
